besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -773,7 +774,9 @@ public void parse(
       final BesuExceptionHandler exceptionHandler,
       final InputStream in,
       final String... args) {
-    commandLine = new CommandLine(this).setCaseInsensitiveEnumValuesAllowed(true);
+    commandLine =
+        new CommandLine(this, new BesuCommandCustomFactory(besuPluginContext))
```

Comment:
Can we just specify the version provider on the commandSpec instead using custom command factory i.e. something like commandLine.getCommandSpec().versionProvider(new VersionProvider(besuPluginContext)

Reply:
Unfortunately not.

Reply:
This approach is suggested in PicoCLI official documentation. https://picocli.info/#_custom_factory